pandas/core/indexes/datetimes.py

```diff
@@ -315,16 +316,18 @@ def _simple_new(cls, values, name=None, freq=None, tz=None, dtype=None):
         if we are passed a non-dtype compat, then coerce using the constructor
         """
         # DatetimeArray._simple_new will accept either i8 or M8[ns] dtypes
-        values = DatetimeArray._simple_new(values, freq=freq, tz=tz)
-
+        if isinstance(values, DatetimeIndex):
+            values = values._data
         dtarr = DatetimeArray._simple_new(values, freq=freq, tz=tz)
+        assert isinstance(dtarr, DatetimeArray)
+
         result = object.__new__(cls)
-        result._data = dtarr._data
-        result._freq = dtarr.freq
+        result._data = dtarr
+        result._freq = dtarr._freq
```

Comment:
there was some trouble related to index._freq not always matching index._data._freq (or possible index.freq vs index._data.freq).  I'm not sure off the top of there is a property keeping them linked.  Even if there is, the `setattr` rules within a `__new__`-like method might be wonky.

Reply:
The index classes themselves shouldn't have a `_freq` attributes. It needs to be a reference to the Array's freq.

Right now it's on DatelikeIndexMixin (which I'd like to remove, and just put those `freq` atts and DatetimeIndexOpsMixin).